### PR TITLE
[OB-3438] fix: ensure context clear on ExitSpace

### PR DESCRIPTION
**Pull request by MAG-MichaelK**
This change ensures that entities are deleted and leader election is reset when exiting a space.

**For the reviewer**

* [ ] If required, are the changes covered by appropriate tests?
* [ ] Are any public-facing API changes well documented?
* [ ] Is the code easily readable and extensible and/or follow existing conventions?

### CODE REVIEW

File: Library/src/Systems/Spaces/SpaceSystem.cpp [hunk 1]
```diff
@@ -260,44 +269,50 @@ void SpaceSystem::ExitSpace(NullResultCallback Callback)
...
-						if (Error != csp::multiplayer::ErrorCode::None)
+						if (Error != multiplayer::ErrorCode::None)
 						{
-						    CSP_LOG_ERROR_FORMAT("Error on exiting spaces whilst clearing scopes, ErrorCode: %s", csp::multiplayer::ErrorCodeToString(Error).c_str());
-                            INVOKE_IF_NOT_NULL(Callback, MakeInvalid<NullResult>());
+							CSP_LOG_ERROR_FORMAT("Error on exiting spaces whilst clearing scopes, ErrorCode: %s",
+												 multiplayer::ErrorCodeToString(Error).c_str());
+							INVOKE_IF_NOT_NULL(Callback, MakeInvalid<NullResult>());
+
+							return;
```
Comment (MAG-SamBirley) [inline Library/src/Systems/Spaces/SpaceSystem.cpp:301]: Nitpick - please could we follow an if/else pattern here rather than earlying out?

**Review by MAG-SamBirley — APPROVED** [review]
Great change! Loads of good app lifecycle hygiene here. 🎉

**Review by MAG-AdamThorn — DISMISSED** [review]

File: Library/src/Systems/Spaces/SpaceSystem.cpp (hunk 1 above)
Comment (MAG-MichaelK) [reply Library/src/Systems/Spaces/SpaceSystem.cpp:301]: We specifically don't do that in most of our code. We always early out for errors.

Comment (MAG-SamBirley) [reply Library/src/Systems/Spaces/SpaceSystem.cpp:301]: That's fair. From a quick scan of `SpaceSystemc.pp` I agree it is used more frequently. Resolving!

**Review by MAG-SamBirley — APPROVED** [review]

**Review by MAG-AdamThorn — APPROVED** [review]

**Pull request merged**